include/polycap.h

```diff
@@ -0,0 +1,125 @@
+	double rad_ext[2],
+	double rad_int[2],
+	double focal_dist[2]); // -> def_cap_profile
+
+// get a new profile from Laszlo's ASCII files
+// perhaps it would be a good idea to define a new filetype that would combine these three into a single file? Best to use something like XML for convenience... this could then be polycap_profile_new_from_xml
+polycap_profile* polycap_profile_new_from_file(
+	const char *single_cap_profile_file,
+	const char *central_axis_file,
+	const char *external_shape_file); // read_cap_profile
+
+// free the polycap_profile structure and its associated data
+void polycap_profile_free(polycap_profile *profile);
+
+// load polycap_description from Laszlo's file. This will recursively call the appropriate polycap_profile_new_* routines. Again here a XML variant could be useful...
+polycap_description* polycap_description_new_from_file(const char *filename);
+
+// get a new polycap_description by providing all its properties... perhaps a simpler variant of this function could be defined that would only set the most important parameters and use defaults for the others??
+polycap_description* polycap_description_new(
+	double sig_rough,
+	double sig_wave
+	double corr_length,
+	double d_source,
+	double d_screen,
+	double src_x,
+	double src_y,
+	double src_sigx,
+	double src_sigy,
+	double src_shiftx,
+	double src_shifty,
+	size_t nelem,
+	int iz[],
+	double wi[],
+	double density,
+	polycap_profile *profile);
+
+// get the polycap_profile from a polycap_description
+polycap_profile* polycap_description_get_profile(polycap_description *description);
+
+// for a given array of energies, and a full polycap_description, get the transmission efficiencies. efficiencies will be allocated by us, and needs to be freed with polycap_free
+int polycap_description_get_transmission_efficiencies(polycap_description *description, size_t n_energies, double *energies, double **efficiencies);
+
+// free a polycap_description struct
+void polycap_description_free(polycap_description *description);
+
+// get a new rng
+polycap_rng* polycap_rng_new(unsigned long int seed);
+
+// free the rng
+void polycap_rng_free(polycap_rng *rng);
+
+// exposing more rng functions may be useful..
+
+// construct a new polycap_photon with its initial position, direction, electric field vector and energy
+polycap_photon* polycap_photon_new(
+	polycap_rng *rng,
+	polycap_vector3 start_coords[3],
+	polycap_vector3 start_direction[3],
+	polycap_vector3 start_electric_vector[3],
+	double energy);
+
+// simulate a single photon for a given polycap_description
+int polycap_photon_launch(polycap_photon *photon, polycap_description *description);
+
+// get exit coordinates
+polycap_vector3 polycap_photon_get_exit_coords(polycap_photon *photon);
+
+// get exit direction
+polycap_vector3 polycap_photon_get_exit_direction(polycap_photon *photon);
+
+// get exit electric vector
+polycap_vector3 polycap_photon_get_exit_electric_vector(polycap_photon *photon);
```

Comment:
wat bedoel je met de electric vector?

Reply:
Is de elektrische vector van het foton geen parameter tijdens de simulatie? Verandert deze niet na een reflectie? In XMI-MSIM is deze van belang, zeker als de exciterende bundel niet polychromatisch is.